aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp

```diff
@@ -411,14 +454,10 @@ namespace Aws
                 m_encryptionMode = false;
                 m_decryptionMode = false;
 
-                if (m_ctx.cipher || m_ctx.cipher_data || m_ctx.engine)
-                {
-                    EVP_CIPHER_CTX_cleanup(&m_ctx);
-                }
-
-                m_ctx.cipher = nullptr;
-                m_ctx.cipher_data = nullptr;
-                m_ctx.engine = nullptr;
+#if OPENSSL_VERSION_NUMBER < 0x10100003L
+                EVP_CIPHER_CTX_cleanup(m_ctx);
+                m_ctx = nullptr;
```

Comment:
After openssl 1.1 m_ctx will be acquired from heap, so here we should free it right?